src/legacy/core_plugins/kibana/public/dashboard/np_ready/dashboard_state_manager.ts

```diff
@@ -229,8 +236,8 @@ export class DashboardStateManager {
 
       if (
         !_.isEqual(
-          convertedPanelStateMap[panelState.explicitInput.id],
-          savedDashboardPanelMap[panelState.explicitInput.id]
+          _.omit(convertedPanelStateMap[panelState.explicitInput.id], 'version'),
+          _.omit(savedDashboardPanelMap[panelState.explicitInput.id], 'version')
         )
```

Comment:
This fixes an edge case with state update when panel with older version than current kibana version is opened before saved with newer version. 
e.g. 7.7.0 panel is opened in master. 

Without this, serialised state technically changes which causes redundant browser history record.  
Change is required to have this test working: https://github.com/elastic/kibana/pull/60087/files#diff-401179577e069eaeb5f20de5ad1ba769R109